services/device-registry-mongodb/src/test/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedCredentialServiceTest.java

```diff
@@ -69,6 +74,9 @@ public void startService(final VertxTestContext testContext) {
 
         vertx = Vertx.vertx();
         mongoClient = MongoDbTestUtils.getMongoClient(vertx, "hono-credentials-test");
+        final DownstreamSenderFactory downstreamSenderFactoryMock = mock(DownstreamSenderFactory.class);
```

Comment:
is this used anywhere in this test?

Reply:
Yes in the line below:

`when(downstreamSenderFactoryMock.connect()).thenReturn(Future.succeededFuture());`

Reply:
Sure, but why should `downstreamSenderFactoryMock.connect()` ever be invoked? I do not see any place where the created downstreamSenderFactoryMock is being used by or injected into any of the objects under test. Or am I mistaken?

Reply:
Sorry - didn't see the wood before the trees. You're right that is obsolete.